ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts

```diff
@@ -0,0 +1,488 @@
+          busterOnYouPuffs: {
+            en: 'Tank Buster on YOU, East/West Between Puffs ',
+          },
+        };
+
+        if (matches.target === data.me) {
+          if (data.beaterCounter === 2)
+            return { alertText: output.busterOnYouPuffs!() };
+          return { infoText: output.busterOnYou!() };
+        }
+
+        if (data.role !== 'tank' && data.role !== 'healer')
+          return;
+
+        return { infoText: output.busterOnTarget!({ player: data.ShortName(matches.target) }) };
+      },
+    },
+    {
+      id: 'ASS Total Wash',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '7750', source: 'Silkie', capture: false }),
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'aoe + bleed',
+          de: 'AoE + Blutung',
+          fr: 'AoE + Saignement',
+          ja: 'AOE + 出血',
+          ko: '전체 공격 + 도트',
+        },
+      },
+    },
+    // ---------------- Gladiator of Sil'dih ----------------
+    {
+      id: 'ASS Flash of Steel',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '7671', source: 'Gladiator of Sil\'dih', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'ASS Rush of Might Collect',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['765C', '765B'], source: 'Gladiator of Sil\'dih' }),
+      preRun: (data, matches) => data.mightCasts.push(matches),
+    },
+    {
+      id: 'ASS Rush of Might 1',
+      // Boss casts 765C (12.2s) and 765B (10.2s), twice
+      // Gladiator of Mirage casts 7659, 7658, 765A, these target the environment but are not reliable
+      // North
+      //                East               West
+      //   Line 1: (-34.14, -270.14) (-35.86, -270.14)
+      //   Line 2: (-39.45, -275.45) (-30.55, -275.45)
+      //   Line 3: (-44.75, -280.75) (-25.25, -280.75)
+      // South
+      //                East               West
+      //   Line 1: (-34.14, -271.86) (-35.86, -271.86)
+      //   Line 2: (-39.45, -266.55) (-30.55, -266.55)
+      //   Line 3: (-44.75, -261.25) (-25.25, -261.25)
+      // Center is at (-35, -271)
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['765C', '765B'], source: 'Gladiator of Sil\'dih', capture: false }),
+      delaySeconds: 0.1,
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        if (data.mightCasts.length !== 4)
+          return;
+
+        const mirage1 = data.mightCasts[0];
+        const unfilteredMirage2 = data.mightCasts[1];
+        const unfilteredMirage3 = data.mightCasts[2];
+
+        if (mirage1 === undefined || unfilteredMirage2 === undefined || unfilteredMirage3 === undefined)
+          throw new UnreachableCode();
+
+        // Filter for unique
+        const mirage2 = (mirage1.x === unfilteredMirage2.x && mirage1.y === unfilteredMirage2.y) ? unfilteredMirage3 : unfilteredMirage2;
```

Comment:
In my logs I am seeing each spell cast twice. There are 4 casts total. We can ignore the last spell since it's just going to be a duplicate of one of the two. I think getCombatants would not work since this is getting the target location of the aoe. The Gladiator Mirages do cast at the target location as well and only twice, but their target location data is what is not reliable. The boss' 4 casts I have seen are reliable.

Reply:
Ah, there is going to be 4 casts because there are two cleaves happening together.

Reply:
Correct, it is two of each ability id. One set at 10.2s cast time, the other at 12.2s.

Reply:
I think it's possible to filter off castTime and just grab one.

Reply:
Ah, it's simpler. The two of one ability id will get both positions.

Reply:
Looking at this more. I see that the data from the boss can sometimes be wrong.
I have a log where the boss targets (-35, -281) for all the casts, which is not correct. Although this seems really rare. I also have a log where the target locations did not change between north and south, which is also wrong

It might be possible to fix this with OverlayPlugin because the boss is targetting one of its own entities, which maybe is invisible and updated before end of cast...

Reply:
Ok I also noticed an oddity, but this is just how the mechanic is, the north/south line 1 here actually does not abide by the < -271 rule. They overlap so a more strict check is needed.

I'll code out a overlayPlugin of this next.

Reply:
This uses overlayPlugin now which will require in game testing to see if it gets the correct values.